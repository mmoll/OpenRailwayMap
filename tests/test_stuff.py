import requests
from lxml import etree
from io import BytesIO

WEB_BASE_URL = 'http://www.openrailwaymap.org'


def test_get_index(snapshot):
    response = requests.get(f'{WEB_BASE_URL}/')

    assert response.status_code == 200

    parser = etree.HTMLParser()
    tree = etree.parse(BytesIO(response.content), parser)
    result = etree.tostring(tree.getroot(), pretty_print=True, method='html')

    assert result == snapshot
