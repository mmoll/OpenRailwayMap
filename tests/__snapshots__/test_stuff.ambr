# name: test_get_index
  b'<html>\n\t<head>\n\t\t<meta charset="utf-8">\n\t\t<meta http-equiv="content-type" content="text/html; charset=UTF-8">\n\t\t<meta http-equiv="content-language" content="en">\n\n\t\t<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">\n\n\t\t<title>OpenRailwayMap</title>\n\t\t<meta name="title" content="OpenRailwayMap">\n\t\t<meta name="application-name" content="OpenRailwayMap">\n\t\t<meta name="page-topic" content="OpenRailwayMap">\n\t\t<meta name="description" content="OpenRailwayMap - An OpenStreetMap-based project for creating a map of the world\'s railway infrastructure.">\n\t\t<meta name="keywords" content="openstreetmap, openrailwaymap, alexander matheisen, rurseekatze, openlayers, osm, matheisen, orm, eisenbahnkarte, bahnkarte, railmap, railway, railways, eisenbahn, streckenkarte">\n\t\t<meta name="robots" content="index,follow">\n\n\t\t<link rel="alternate" type="application/rss+xml" title="OpenRailwayMap Deutsch" href="https://blog.openrailwaymap.org/de.rss" hreflang="de">\n\t\t<link rel="alternate" type="application/rss+xml" title="OpenRailwayMap English" href="https://blog.openrailwaymap.org/en.rss" hreflang="en">\n\n\t\t<link rel="search" type="application/opensearchdescription+xml" href="opensearch.xml" title="OpenRailwayMap">\n\n\t\t<link rel="shortcut icon" href="img/openrailwaymap-16.png" type="image/png">\n\t\t<link rel="icon" href="img/openrailwaymap-16.png" type="image/png">\n\n\t\t<meta http-equiv="content-script-type" content="text/javascript">\n\t\t<meta http-equiv="content-style-type" content="text/css">\n\n\t\t<link rel="stylesheet" type="text/css" href="css/map.css">\n\t\t<link rel="stylesheet" type="text/css" href="css/leaflet.css">\n\n\t\t<script type="text/javascript" src="js/leaflet.js"></script>\n\t\t<link rel="stylesheet" href="css/Leaflet.EditInOSM.css">\n\t\t<script src="js/Leaflet.EditInOSM.js"></script>\n\t\t<script type="text/javascript" src="js/L.TileLayer.Grayscale.js"></script>\n\t\t<script type="text/javascript">\nvar params={\nurlbase : \'http://dev.openrailwaymap.org//\',\nid : null,\ntype : null,\nlat : null,\nlon : null,\nzoom : null,\nlang : null,\noffset : null,\nsearchquery : \'\',\nref : null,\nname : null,\nline : null,\noperator : null,\nmobile : null,\nstyle : null\n};\n</script>\n\t\t<script type="text/javascript" src="js/search.js"></script>\n\t\t<script type="text/javascript" src="js/startposition.js"></script>\n\t\t<script type="text/javascript" src="js/timestamp.js"></script>\n\t\t<script type="text/javascript" src="js/functions.js"></script>\n\t\t<script type="text/javascript" src="js/bitmap-map.js"></script>\n\t\t<!-- redirect to mobile version if necessary -->\n\t\t<script type="text/javascript">\n\t\t\tmobileRedirection();\n\t\t</script>\n\n\t\t<meta name="mobile-web-app-capable" content="yes">\n\t\t<meta name="apple-mobile-web-app-capable" content="yes">\n\t\t<meta name="apple-mobile-web-app-title" content="OpenRailwayMap">\n\t\t<link rel="manifest" href="webapp-manifest.json">\n\t\t<meta name="theme-color" content="#4e9a06">\n\n\t\t<meta name="twitter:card" content="summary">\n\t\t<meta name="twitter:site" content="@openrailwaymap">\n\t\t<meta name="twitter:title" content="OpenRailwayMap">\n\t\t<meta name="twitter:description" content="OpenRailwayMap - An OpenStreetMap-based project for creating a map of the world&amp;#39;s railway infrastructure.">\n\t\t<meta name="twitter:url" content="https://www.openrailwaymap.org/">\n\t\t<meta name="twitter:image" content="https://www.openrailwaymap.org/img/openrailwaymap-310.png">\n\t</head>\n\t<body onload="createMap(false);">\n\t\t<div id="moreInfo" class="moreInfoFalse"></div>\n\t\t<div id="sideBar" class="sideBar">\n\t\t\t<b id="header"><a href="index.php">OpenRailwayMap</a></b>\n\t\t\t<form id="langSelection">\n\t\t\t\t<select id="langSelector" size="1" onchange="changeLanguage(gEBI(\'langSelector\').options[gEBI(\'langSelector\').selectedIndex].value)">\n\t\t\t\t\t<option value="ca">Catal&#224;</option>\n<option value="cs">&#268;esky</option>\n<option value="da">Dansk</option>\n<option value="de">Deutsch</option>\n<option value="el">&#917;&#955;&#955;&#951;&#957;&#953;&#954;&#940;</option>\n<option value="en" selected>English</option>\n<option value="es">Espa&#241;ol</option>\n<option value="fi">Suomi</option>\n<option value="fr">Fran&#231;ais</option>\n<option value="hu">Magyar</option>\n<option value="ja">&#26085;&#26412;&#35486;</option>\n<option value="lt">Lietuvi&#371;</option>\n<option value="nl">Nederlands</option>\n<option value="nqo">&#2002;&#2014;&#1999;</option>\n<option value="pl">Polski</option>\n<option value="pt">Portugu&#234;s</option>\n<option value="ru">&#1056;&#1091;&#1089;&#1089;&#1082;&#1080;&#1081;</option>\n<option value="sl">Sloven&#353;&#269;ina</option>\n<option value="sv">Svenska</option>\n<option value="tr">T&#252;rk&#231;e</option>\n<option value="uk">&#1059;&#1082;&#1088;&#1072;&#1111;&#1085;&#1089;&#1100;&#1082;&#1072;</option>\n<option value="vi">Ti&#7871;ng Vi&#7879;t</option>\n<option value="zh">&#27491;&#39636;&#20013;&#25991;</option>\n\t\t\t\t</select>\n\t\t\t</form>\n\t\t\t<br>\n\t\t\t<p id="info"></p>\n\t\t\t<div id="linkBar">\n\t\t\t\t<a class="links" id="infoButton" href="https://wiki.openstreetmap.org/wiki/OpenRailwayMap" target="_blank">More Info</a>&#160;&#8226;\n\t\t\t\t<a class="links" id="infoButton" href="https://blog.openrailwaymap.org/" target="_blank">Blog</a>&#160;&#8226;\n\t\t\t\t<a class="links" id="contactButton" href="#">Contact</a>&#160;&#8226;\n\t\t\t\t<a class="links" id="paypalButton" href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=9KCKT39N7AGL8">Donate</a>&#160;&#8226;\n\t\t\t\t<a class="links" id="imprintButton" href="/imprint">Imprint &amp; Privacy Policy</a>&#160;&#8226;\n\t\t\t\t<a class="links" id="permalinkButton" href="">Permalink</a>\n\t\t\t\t<script language="javascript">\n\t\t\t\t\tvar usr = "info";\n\t\t\t\t\tvar dom = "openrailwaymap";\n\t\t\t\t\tvar tld = "org";\n\t\t\t\t\tgEBI("contactButton").href="mailto:"+usr+"@"+dom+"."+tld;\n\t\t\t\t</script>\n\t\t\t</div>\n\t\t\t<input type="text" id="searchBox" size="20">\n\t\t\t<div id="clearButton" onclick="Search.clear();">\n\t\t\t\t<b>&#10060;</b>\n\t\t\t</div>\n\t\t\t<div id="searchButton" onclick="Search.request();">\n\t\t\t\t<b>&#8629;</b>\n\t\t\t</div>\n\t\t\t<br>\n\t\t\t<div id="searchBar" class="infoBarOut"></div>\n\t\t\t<div id="detailsBar" class="infoBarOut"></div>\n\t\t\t<div id="styleSelectionBar"></div>\n\t\t\t<div id="legendBar">\n\t\t\t\t<b>Legend:</b>\n\t\t\t\t<iframe id="legend" scrolling="no" onload="setIframeHeight(this.id)" src=""></iframe>\n\t\t\t</div>\n\t\t\t<iframe id="josmFrame" src="about:blank"></iframe>\n\t\t</div>\n\t\t<div id="locateButton"></div>\n\t\t<div id="mapFrame">\n\t\t\t<noscript>\n\t\t\t\t<p><b>Javascript is not activated</b><br>Javascript is needed to show the map and run this website. Please turn on Javascript in your browser settings.</p>\n\t\t\t</noscript>\n\t\t</div>\n\t</body>\n</html>\n'
---
